Lib/test/test_syntax.py

```diff
@@ -2117,6 +2117,7 @@ def test_syntax_error_on_deeply_nested_blocks(self):
         self._check_error(source, "too many statically nested blocks")
 
     @support.cpython_only
+    @unittest.skipIf(support.is_wasi, "Exhausts WASI call stack")
```

Comment:
Is there some way we could tweak the test to get the same coverage?

Reply:
I was able to make the test pass with a smaller parser ``MAXSTACK`` value on WASI.

CC @pablogsal